client/my-sites/domains/domain-management/transfer/index.jsx

```diff
@@ -25,35 +25,47 @@ import {
 import VerticalNav from 'components/vertical-nav';
 import VerticalNavItem from 'components/vertical-nav/item';
 import getCurrentRoute from 'state/selectors/get-current-route';
+import { getSelectedDomain, isMappedDomain } from 'lib/domains';
+import DomainMainPlaceholder from 'my-sites/domains/domain-management/components/domain/main-placeholder';
 
 function Transfer( props ) {
 	const {
 		isAtomic,
 		isDomainOnly,
+		isMappedDomain,
 		isPrimaryDomain,
 		selectedSite,
 		selectedDomainName,
 		currentRoute,
 		translate,
 	} = props;
-
 	const slug = get( selectedSite, 'slug' );
 
+	if ( props.isRequestingSiteDomains ) {
+		return (
+			<DomainMainPlaceholder
+				backHref={ domainManagementEdit( slug, selectedDomainName, currentRoute ) }
+			/>
+		);
+	}
+
 	return (
 		<Main>
 			<Header
 				selectedDomainName={ selectedDomainName }
 				backHref={ domainManagementEdit( slug, selectedDomainName, currentRoute ) }
 			>
-				{ translate( 'Transfer Domain' ) }
+				{ ! isMappedDomain ? translate( 'Transfer Domain' ) : translate( 'Transfer Mapping' ) }
```

Comment:
Minor nit: I would find it easier to read here and elsewhere if we flipped this and used `isMappedDomain ? translate( 'Transfer Mapping' ) : translate( 'Transfer Domain' )`. But this is definitely not a blocker, just a minor personal preference.

Reply:
Yeah, I was wondering about that myself - went with the inverted originally to keep the "main" branch the most likely one... but it does make it slightly harder to understand (and easier to misunderstand) so I'll flip it.